debrid/services/realdebrid.py

```diff
@@ -189,9 +189,26 @@ def download(element, stream=True, query='', force=False):
                                     delete('https://api.real-debrid.com/rest/1.0/torrents/delete/' + torrent_id)
                                     continue
                         else:
-                            ui_print(f'[realdebrid]: {release.title} is in status [{response.status}] - trying a different release.')
-                            delete('https://api.real-debrid.com/rest/1.0/torrents/delete/' + torrent_id)
-                            continue
+                            if response.status in ["waiting_files_selection"]:
+                                if hasattr(response, "files") and len(response.files) > 0:
+                                    version_files = []
+                                    for file_ in response.files:
+                                        debrid_file = file(file_.id, file_.path, file_.bytes, release.wanted_patterns, release.unwanted_patterns)
+                                        version_files.append(debrid_file)
+                                    release.files = [version(version_files)]
+                                    cached_ids = [vf.id for vf in version_files if vf.wanted and not vf.unwanted and vf.name.endswith(tuple(media_file_extensions))]
+                                    if len(cached_ids) == 0:
+                                        ui_print('[realdebrid] no selectable media files.', ui_settings.debug)
+                                    else:
+                                        post('https://api.real-debrid.com/rest/1.0/torrents/selectFiles/' + torrent_id, {'files': ",".join(map(str, cached_ids))})
+                                        ui_print(f'[realdebrid]: {release.title} is in status [{response.status}] - tried start download of torrent by selecting files.')
+                                        ui_print('[realdebrid] selectFiles response ' + repr(response), ui_settings.debug)
+                                continue
+                            else:
+                                ui_print(f'[realdebrid]: {release.title} is in status [{response.status}] - trying a different release.')
+                                delete('https://api.real-debrid.com/rest/1.0/torrents/delete/' + torrent_id)
+                                continue
+                            
                     if response.status == 'downloaded':

```

Comment:
Hi, is this not covered on lines 157:168 above? If it gets to this point, then we've already confirmed there are no selectable media files. Whether it's in the status `waiting_files_selection` is irrelevant at this point because we intentionally did not select any files above. ie. `len(cached_ids) == 0` will always be true here since we've already confirmed this above on line 164?

Reply:
@sirstudly  I have not debugged that section properly. But while running I was getting too many torrents  in status waiting_files_selection . So tried adding this which resovled issue for me and more torrents got downloaded.

Might need more debugging on line 163, why file selection is getting skipped in there.

Reply:
I'm happy to look into this further but are you able to give me an example where this change downloads a release that would otherwise have been skipped? Is it likely to be a tv show or movie? Line 163 may give an empty list for shows if the files inside the release aren't properly named (eg. S03E14).. But this would occur on your line 199 as well since it's the same code so i'm not sure how this change could give you better results...